library/std/src/sys/pal/xous/thread_local_key.rs

```diff
@@ -91,8 +91,7 @@ fn tls_table() -> &'static mut [*mut u8] {
 #[inline]
 pub unsafe fn create(dtor: Option<Dtor>) -> Key {
     // Allocate a new TLS key. These keys are shared among all threads.
-    #[allow(unused_unsafe)]
```

Comment:
This is necessary because when running tests, `TLS_KEY_INDEX` is an extern, and so needs to be accessed with `unsafe`.

It seems to be possible to accomplish the same thing with `#[cfg_attr(test, linkage = "available_externally")]` as sgx appears to do. This clears up a lot of the ugliness in the file with respect to the two variables that must be shared.

Thank you for the heads-up. Now that I'm aware of that trick, I'll apply it in a future patch that fixes this on Xous.